WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -1136,28 +1136,24 @@ private extension ProductFormViewController {
             present(blazeHostingController, animated: true)
             ServiceLocator.analytics.track(event: .Blaze.blazeEntryPointDisplayed(source: .introView))
         } else {
-            navigateToBlazeCampaignCreation(siteUrl: site.url, source: .productDetailPromoteButton)
+            navigateToBlazeCampaignCreation(siteID: site.siteID, siteUrl: site.url, source: .productDetailPromoteButton)
         }
     }
 
-    private func navigateToBlazeCampaignCreation(siteUrl: String, source: BlazeSource) {
-        if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.blazei3NativeCampaignCreation) {
-            let viewModel = BlazeCampaignCreationFormViewModel(siteID: viewModel.productModel.siteID,
-                                                               productID: viewModel.productModel.productID) { [weak self] in
-                self?.handlePostCreation()
-            }
-            let controller = BlazeCampaignCreationFormHostingController(viewModel: viewModel)
-            navigationController?.show(controller, sender: self)
-        } else {
-            let blazeViewModel = BlazeWebViewModel(siteID: viewModel.productModel.siteID,
-                                                   source: source,
-                                                   siteURL: siteUrl,
-                                                   productID: product.productID) { [weak self] in
-                self?.handlePostCreation()
-            }
-            let webViewController = AuthenticatedWebViewController(viewModel: blazeViewModel)
-            navigationController?.show(webViewController, sender: self)
+    private func navigateToBlazeCampaignCreation(siteID: Int64, siteUrl: String, source: BlazeSource) {
+        guard let navigationController = self.navigationController else {
+            return
         }
+
+        let coordinator = BlazeCampaignCreationCoordinator(
+            siteID: siteID,
+            siteURL: siteUrl,
+            productID: product.productID,
+            source: source,
+            navigationController: navigationController,
+            onCampaignCreated: handlePostCreation
+        )
+        coordinator.start()
```

Comment:
We should keep a strong reference to the coordinator in the class just in case this local variable gets deallocated.

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/11671/commits/fe0fd17eedb2a8a24806451cdc59a570e1a2d04e